### PR TITLE
Improve addons selector dialog

### PR DESCRIPTION
**Pull request by dgdavid**
### Problem

[The dialog](https://github.com/yast/yast-registration/blob/9c04bc3d0c44488425f056f781d0c091a9efb10e/src/lib/registration/ui/addon_selection_base_dialog.rb) allowing to select addons to be installed presents at least two problems:

* resets the scroll, when present, after each selection
* forces to select/unselect (or vice-versa) the item to see its details

_To read more see the #483 description and https://github.com/yast/yast-registration/pull/483#discussion_r385108398_

Related to 
  - https://trello.com/c/CmaNrDAw
  - https://bugzilla.suse.com/show_bug.cgi?id=1049965
  - https://bugzilla.suse.com/show_bug.cgi?id=1161200

### Solution

Implements a new dialog to manage the addons selection, relying in [the new  `CWM::MultiStatusSelector` custom widget](https://github.com/yast/yast-yast2/pull/1029).

This new widget fixes mentioned problems without the `MultiSelectionBox` limitations.

### Tests

* Added unit tests
* Also tested manually during installation via [`yupdate`](https://github.com/yast/yast-installation/blob/master/doc/yupdate.md)

### Screenshots

<details>
<summary>Click to show/hide the GUI result</summary>

---

#### Arriving to the addons selector dialog

Screenshot_sl5sp2_2020-03-13_13:17:07

#### Displaying an auto selected module

Screenshot_sl5sp2_2020-03-13_13:17:21

#### Displaying an addons details by clicking in its label, without getting it selected

Screenshot_sl5sp2_2020-03-13_13:17:29

#### Displaying development versions

Screenshot_sl5sp2_2020-03-13_13:17:39

#### Displaying the help text

Screenshot_sl5sp2_2020-03-13_13:19:03
</details>

<details>
<summary>Click to show/hide screenshots when running in textmode</summary>

---

#### Arriving to the addons selector dialog

Screenshot_sl5sp2_2020-03-13_13:26:09

#### Displaying an auto selected module

Screenshot_sl5sp2_2020-03-13_13:26:36

#### Displaying development versions

Screenshot_sl5sp2_2020-03-13_13:27:16

#### Displaying the help text

Screenshot_sl5sp2_2020-03-13_13:28:39
</details>

### CODE REVIEW

File: src/lib/registration/widgets/addons_selector.rb [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+        end
+
+        # @macro seeAbstractWidget
+        def init
+          self.value = @initial
+        end
+
+        # @macro seeAbstractWidget
+        def label
+          _("&Hide Development Versions")
```
Comment (lslezak) [inline src/lib/registration/widgets/addons_selector.rb]: Please add a `#TRANSLATORS: check box label` comment.

File: src/lib/registration/widgets/addons_selector.rb [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+        # Builds and return the item label based on the addon label and availability
+        #
+        # @return [String] the item label
+        def label
+          @label ||=
+            if addon.available?
+              addon.label
+            else
+              _("%s (not available)") % addon.label
```
Comment (lslezak) [inline src/lib/registration/widgets/addons_selector.rb]: Add `# TRANSLATORS: label for a not available module/extension, %s is replaced by the module/extension name` comment

**Comment by dgdavid** [comment]
@lslezak 

I have been testing this with the latest SUSE full installation media, and it works fine, even going back and forwar.

The only two _minor_ weird things, as you can see in the uploaded screenshots, are 

* those two missing icons in the help text (I don't know why, since [it is the same than used for items](https://github.com/yast/yast-yast2/blob/7c58df87a0c9f2c8cdbcb8f935b3c6e45c84f152/library/cwm/src/lib/cwm/multi_status_selector.rb#L321-L332))
* an extra white line when the item label is as long as the available area, which I think is a RichText wrapping problem :man_shrugging:

File: src/lib/registration/widgets/addons_selector.rb [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+
+module Registration
+  module Widgets
+    # The addons multi selector widget, capable to deal with both
+    #
+    #   - the state (enable or disable)
+    #   - the status (selected/registered, not selected, or auto selected)
+    #
+    # @see CWM::MultiStatusSelector
+    class AddonsSelector < CWM::MultiStatusSelector
```
Comment (joseivanlopez) [inline src/lib/registration/widgets/addons_selector.rb]: I haven't follow the discussion in this PR, but I would say that `AddonsSelector` should be a `CustomWidget` containing a `MultiStatusSelector` internally.

Comment (joseivanlopez) [reply src/lib/registration/widgets/addons_selector.rb]: In other words, using composition instead of inheritance from `MultiStatusSelector`.

Comment (dgdavid) [reply src/lib/registration/widgets/addons_selector.rb]: Thanks for the review @joseivanlopez

Reading the comment, probably you're right. The problem is that I did it with the inheritance in mind, according with the agreed in https://gist.github.com/joseivanlopez/f520017cb0effe70887a4447562b0339#file-multi_state_selection_box-rb-L73

I'm not against in changing it for improving it, but this task already consumed a lot of time and the team is waiting for it to make the SP2 branching.

So, if it does not hurt too much, I think we should go ahead postponing the improvements for a next iterations.

**Pull request merged**